nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -937,6 +938,31 @@ private void setNullnessForMapCalls(
         Nullness value = inputs.valueOfSubNode(arguments.get(1));
         bothUpdates.set(getAccessPath, value);
       }
+    } else if (AccessPath.isMapComputeIfAbsent(callee, state)) {
+      AccessPath getAccessPath = AccessPath.getForMapInvocation(node, apContext);
+      if (getAccessPath != null) {
+        // TODO: For now, Function<K, V> implies a @NonNull V. We need to revisit this once we
+        // support generics, but we do include a couple defensive tests below.
+        if (arguments.size() < 2) {
+          return;
+        }
+        Node funcNode = arguments.get(1);
+        if (!funcNode.getType().getKind().equals(TypeKind.DECLARED)) {
+          return;
+        }
+        Type.ClassType classType = (Type.ClassType) funcNode.getType();
+        if (classType.getTypeArguments().size() != 2) {
+          return;
+        }
+        Type functionReturnType = classType.getTypeArguments().get(1);
+        // Unfortunately, functionReturnType.tsym seems to elide annotation info, so we can't call
+        // the Nullness.* methods that deal with Symbol. We might have better APIs for this kind of
+        // check once we have real generics support.
+        if (!Nullness.hasNullableAnnotation(
+            functionReturnType.getAnnotationMirrors().stream(), config)) {
```

Comment:
Presumably we could keep adding checks here, including checking explicitly for a lambda and querying its own dataflow results, but there would still be other edge-cases (method references, for example), and the real way to handle is that `Map<K!, V!>` has the method `Map.computeIfAbsent(K! k, Function!<K!, V!>)` and `Map<K!, V?>` should probably just return `@Nullable` from get (i.e. `V? Map.get(K!)`) regardless of which paths have been set or not... but that needs generics support.

Reply:
I think what you've implemented is completely reasonable for now.